runtime/lua/vim/ui.lua

```diff
@@ -179,7 +179,13 @@ function M._get_url()
       current_node = current_node:parent()
     end
   end
-  return vim.fn.expand('<cfile>')
+
+  local old_isfname = vim.o.isfname
+  vim.cmd [[set isfname+=@-@]]
+  local url = vim.fn.expand('<cfile>')
+  vim.o.isfname = old_isfname
```

Comment:
off topic: we could introduce an internal/private `vim._with()` https://github.com/neovim/neovim/issues/19832 to start discovering places where it's useful and figure out its interface.

Reply:
https://github.com/neovim/neovim/pull/29244